tests/test_mock_sampler.py

```diff
@@ -216,7 +217,90 @@ def test_yield_arguments(self):
                                    broken_edges=delete_edges)
         self.assertTrue(len(sampler.nodelist)==4)
         self.assertTrue(len(sampler.edgelist)==1)
-        
+
+    def test_custom_substitute_sampler(self):
+        """Test that MockDWaveSampler uses the provided custom substitute_sampler."""
+
+        # Define a sampler that always returns the a constant (excited) state
+        class SteepestAscentSolver(SteepestDescentSolver):
+            def __init__(self):
+                super().__init__()
+
+            def sample(self, bqm, **kwargs):
+                # Return local (or global)  maxima instead of local minima
+                return super().sample(-bqm, **kwargs)
+
+        inverted_sampler = SteepestAscentSolver()
+
+        # Create a simple BQM
+        bqm = dimod.BQM({'a': 1, 'b': 1}, {}, 0.0, vartype="SPIN")
+
+        # Instantiate MockDWaveSampler with nodelist and edgelist including 'a' and 'b'
+        sampler = MockDWaveSampler(
+            substitute_sampler=inverted_sampler,
+            nodelist=['a', 'b'],
+            edgelist=[('a', 'b')]
+        )
+
+        # First sample does not use ExactSampler(); Second sample does not
+        # use SteepestDescentSampler()
+        ss = sampler.sample(bqm, num_reads=2)
+        self.assertEqual(sampler.exact_solver_cutoff, 0)
+        self.assertEqual(ss.record.sample.shape, (1,2), 'Unique sample expected')
```

Comment:
```suggestion
        self.assertEqual(ss.aggregate().record.sample.shape, (1,2), 'Unique sample expected')
```
or remove, as this case is covered by the other tests